codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2LocalFlowController.java

```diff
@@ -164,6 +161,18 @@ public int unconsumedBytes(Http2Stream stream) {
         return state(stream).unconsumedBytes();
     }
 
+    @Override
+    public void flushWindowUpdates(final ChannelHandlerContext ctx) throws Http2Exception {
+        connectionState().writeWindowUpdateIfNeeded(ctx);
+        connection.forEachActiveStream(new Http2StreamVisitor() {
```

Comment:
I doubt it's really affecting performance, most connections will have a low number of streams and if there's nothing to do for a stream it will exit out pretty quickly anyway.